### ISSUE TITLE
Creating more tests 

### ISSUE THREAD

**Issue by huguesdevimeux** [ManimCommunity/manim#193]
At the moment we only have last-frames tests. 

Although they are useful, it's not enough. 

We would need to have tests for cli flags, that check videos, etc. 

Tell your ideas here !

**Comment by leotrs** [comment]
We should have at least one test for each CLI flag.

**Comment by leotrs** [comment]
We should test that the cascading priorities in the config system work as expected.

**Comment by eulertour** [comment]
Ties in to #11.

**Comment by naveen521kk** [comment]
I recently came up with some good things like [coverage.py](https://coverage.readthedocs.io/en/coverage-5.3/) which is great. What it actually does is checks how many parts of code is executed when the tests are running and how many doesn't(for example checks how many if condition code are not executed as so on). It make beautiful html report or we can something like https://codecov.io for managing that. Pretty much, it will help us determine how good our tests are and to prevent code breakage. I suggest we should set it up soon and it isn't hard to it.

**Comment by leotrs** [comment]
Before looking into code coverage tools, I would like to see more tests.

**Comment by csrabak** [comment]
@leotrs :

The two subjects are well interconnected, we could say they're conjoined twins ‼️ 

Adding tests if do not exercise all the control flow paths may give a sense of security when in fact are only redundancies.

That's why coverage should be considered as integral part of the testing exercise.

my .01999....

**Comment by huguesdevimeux** [comment]
> That's why coverage should be considered as integral part of the testing exercise.

I agree with that

**Comment by naveen521kk** [comment]
Probably we can open a new issue for that. @huguesdevimeux

**Comment by leotrs** [comment]
I agree that coverage should be taken into account, I just meant to say that right now the priority should go to writing tests. If anyone is looking into implementing coverage, that time would 100% be better spent by implementing more tests, as of right now.

Having said that, if anyone wants to do that now, then by all means. But deciding upon one more tool should be discussed in a separate issue, and we should talk about a tool that works for doctests as well, at least.

**Comment by eulertour** [comment]
I just wanted to add that testing frames in the middle of each animation would be very helpful, since a lot of the changes I've been making have passed the tests even if they don't actually render the video properly. The only solution right now is to constantly run several animations and hope they offer enough coverage.

**Comment by lucasricci** [comment]
@huguesdevimeux  I would like to take the creation of a test for ``add_fixed_in_frame_mobjects``. Of course, if still available.

**Comment by huguesdevimeux** [comment]
> @huguesdevimeux  I would like to take the creation of a test for ``add_fixed_in_frame_mobjects``. Of course, if still available.

Sure !

**Comment by giolucasd** [comment]
I would like to add unit tests for utility on "utils/polylabel.py", because it is used a lot and is only tested indirectly. It also seems easier for beginers. Any objections?
